stdlib/public/core/Arrays.swift.gyb

```diff
@@ -2021,31 +2021,38 @@ extension _ArrayBufferProtocol {
   internal mutating func _arrayAppendSequence<S : Sequence>(
     _ newItems: S
   ) where S.Iterator.Element == Element {
+    
+    // this function is only ever called from append(contentsOf:)
+    // which should always have exhausted its capacity before calling
+    _sanityCheck(count == capacity)
+    var newCount = self.count
 
+    // there might not be any elements to append remaining,
+    // so use an iterator+while loop to check, then increase
```

Comment:
"use an iterator+while loop" is the sort of thing we shouldn't put in code comments; it just says what is very plainly obvious.